diff --git a/src/codegen/llvm/FuncGen.zig b/src/codegen/llvm/FuncGen.zig
index 40fc1f56a8..3b9253142b 100644
--- a/src/codegen/llvm/FuncGen.zig
+++ b/src/codegen/llvm/FuncGen.zig
@@ -2127,7 +2127,7 @@ fn airSliceElemVal(self: *FuncGen, inst: Air.Inst.Index) Allocator.Error!Builder
     if (isByRef(elem_ty, zcu)) {
         return self.loadByRef(ptr, elem_ty, elem_align.toLlvm(), access_kind);
     } else {
-        return self.loadTruncate(access_kind, elem_ty, ptr, elem_align.toLlvm());
+        return self.wip.load(access_kind, try self.object.lowerType(elem_ty), ptr, elem_align.toLlvm(), "");
     }
 }
 
@@ -2157,7 +2157,7 @@ fn airArrayElemVal(self: *FuncGen, inst: Air.Inst.Index) Allocator.Error!Builder
             const elem_align = elem_ty.abiAlignment(zcu).toLlvm();
             return self.loadByRef(elem_ptr, elem_ty, elem_align, .normal);
         } else {
-            return self.loadTruncate(.normal, elem_ty, elem_ptr, .default);
+            return self.wip.load(.normal, try self.object.lowerType(elem_ty), elem_ptr, .default, "");
         }
     }
 
@@ -2277,7 +2277,7 @@ fn airStructFieldVal(self: *FuncGen, inst: Air.Inst.Index) Allocator.Error!Build
     if (isByRef(field_ty, zcu)) {
         return self.loadByRef(field_ptr, field_ty, field_ptr_align.toLlvm(), .normal);
     } else {
-        return self.loadTruncate(.normal, field_ty, field_ptr, field_ptr_align.toLlvm());
+        return self.wip.load(.normal, try self.object.lowerType(field_ty), field_ptr, field_ptr_align.toLlvm(), "");
     }
 }
 
@@ -6172,7 +6172,7 @@ fn optPayloadHandle(
         if (can_elide_load) return payload_ptr;
         return fg.loadByRef(payload_ptr, payload_ty, payload_align, .normal);
     } else {
-        return fg.loadTruncate(.normal, payload_ty, payload_ptr, payload_align);
+        return fg.wip.load(.normal, try fg.object.lowerType(payload_ty), payload_ptr, payload_align, "");
     }
 }
 
@@ -6269,7 +6269,7 @@ fn load(
     if (isByRef(elem_ty, zcu)) {
         return fg.loadByRef(ptr, elem_ty, ptr_alignment, access_kind);
     } else {
-        return fg.loadTruncate(access_kind, elem_ty, ptr, ptr_alignment);
+        return fg.wip.load(access_kind, try fg.object.lowerType(elem_ty), ptr, ptr_alignment, "");
     }
 }
 
